src/mjolnir/pbfgraphparser.cc

```diff
@@ -2226,10 +2229,14 @@ void PBFGraphParser::ParseNodes(const boost::property_tree::ptree& pt,
   // using much mem, the scoping makes sure to let it go when done sorting
   LOG_INFO("Sorting osm way node references by node id...");
   {
+    sequence<OSMWayNode> way_nodes_tmp(way_nodes_tmp_file, true);
     sequence<OSMWayNode> way_nodes(way_nodes_file, false);
-    way_nodes.sort(
-        [](const OSMWayNode& a, const OSMWayNode& b) { return a.node.osmid_ < b.node.osmid_; });
+    way_nodes.merge_sort(
+        [](const OSMWayNode& a, const OSMWayNode& b) { return a.node.osmid_ < b.node.osmid_; }, way_nodes_tmp);
   }
+  LOG_INFO("Merge sort done, now swapping files.");
+  filesystem::remove(way_nodes_file);
+  filesystem::rename(way_nodes_tmp_file, way_nodes_file);
```

Comment:
@kevinventullo what do you think about encapsulating the whole, make a new sequence/file, do the sort, remove and rename the file back, into the `merge_sort` function, so that instead of passing it a sequence and hanlding it outside you pass in the tmp file location and let all that stuff encapsulated inside the function?

Reply:
i made the chagnes locally to move the creation of a temporary file into the merge_sort function. ill verify its working and then push it up